src/Beman/Optional26/tests/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+# cmake-format: off
+# src/Beman/Optional26/tests/CMakeLists.txt -*-makefile-*-
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# cmake-format: on
+
+# Tests
+add_executable(optional_test "")
+
+target_sources(
+  optional_test
+  PRIVATE optional.t.cpp optional_ref.t.cpp
+          optional_monadic.t.cpp optional_range_support.t.cpp
+          optional_ref_monadic.t.cpp detail/iterator.t.cpp)
+
+target_link_libraries(optional_test "${TARGET_LIBRARY}" GTest::gtest GTest::gtest_main)
+
+include(GoogleTest)
+
+# Note: clang-19 + gtest_discover_tests + Asan setup causes errors on some platforms.
+# Temporary switch to gtest_add_tests and skip some Asan checks.
```

Comment:
Did you consider creating a GitHub issue to track this and linking to it from here? That would allow for future maintainers to know when a workaround is no longer necessary.

Also, the comment mentions a temporary switch to gtest_add_tests, but it isn't clear what the preferred alternative is. Additionally I don't see any code below mentioning ASAN.

Reply:
> Did you consider creating a GitHub issue to track this and linking to it from here? That would allow for future maintainers to know when a workaround is no longer necessary.

Thanks for suggestion! Created #32 .

> so, the comment mentions a temporary switch to gtest_add_tests, but it isn't clear what the preferred alternative is. Additionally I don't see any code below mentioning ASAN.

Please check https://github.com/beman-project/Optional26/blob/main/.github/workflows/ci.yml#L78. We are running with ASAN enabled on CI.

Reply:
I think we still need the link in the comment above to the issue.

Reply:
done